composer/utils/checkpoint.py

```diff
@@ -621,7 +621,16 @@ def load_sharded_checkpoint(
             # 1. Load metadata first for backwards compatability check
             # We need to check if the "optimizers" is at the root of the state dict to determine
             # how to load the optimizer state.
-            metadata = storage_reader.read_metadata()
+            try:
+                metadata = storage_reader.read_metadata()
+            except AttributeError as e:
+                if '_MEM_FORMAT_ENCODING' in str(e):
+                    raise ValueError(
+                        'Unable to read checkpoint metadata. The checkpoint was likely saved with a '
+                        'newer version of torch. Upgrade your torch version to load this checkpoint.',
+                    )
+                else:
+                    raise
```

Comment:
Uhh this is a bug right? You need to have `raise e`?

Reply:
you can also do `raise` like this to reraise the error

Reply:
nah
<img width="313" alt="Screenshot 2024-05-06 at 2 38 43 PM" src="https://github.com/mosaicml/composer/assets/28932043/3f5e1e39-1772-41e6-b6fe-2957662b113a">